Pal/src/host/Linux/db_exception.c

```diff
@@ -207,23 +207,33 @@ static bool _DkGenericSignalHandle (int event_num, siginfo_t * info,
  */
 static struct pal_frame * get_frame (ucontext_t * uc)
 {
-    unsigned long rip = uc->uc_mcontext.gregs[REG_RIP];
-    unsigned long rbp = uc->uc_mcontext.gregs[REG_RBP];
-    unsigned long last_rbp = rbp - 64;
+    uintptr_t rip = uc->uc_mcontext.gregs[REG_RIP];
+    uintptr_t rbp = uc->uc_mcontext.gregs[REG_RBP];
 
     if (!ADDR_IN_PAL(rip))
         return NULL;
 
-    while (ADDR_IN_PAL(((unsigned long *) rbp)[1])) {
+    return __get_frame(rbp);
+}
+
+struct pal_frame * __get_frame (uintptr_t rbp)
+{
+    if (!rbp)
+        asm volatile ("movq %%rbp, %0" : "=r"(rbp) :: "memory");
+
+    uintptr_t last_rbp = rbp - 64;
+    while (ADDR_IN_PAL(((uintptr_t *) rbp)[1])) {
         last_rbp = rbp;
-        rbp = *(unsigned long *) rbp;
+        rbp = *(uintptr_t *) rbp;
```

Comment:
It might be worth a comment that we require the PAL to use frame pointers, and that we don't expect this to ever happen in the bottom-most PAL frame.